sparta/sparta/kernel/Scheduler.hpp

```diff
@@ -931,36 +931,36 @@ class Scheduler : public RootTreeNode
     std::unique_ptr<DAG> dag_;
 
     //! The number of groups in the DAG after finalization.
-    uint32_t dag_group_count_;
+    uint32_t dag_group_count_ = 1;
 
     //! The number of firing groups (dag_group_count_ [+pre] [+post])
-    uint32_t firing_group_count_;
+    uint32_t firing_group_count_ = {dag_group_count_ + 2};
```

Comment:
Magic number...why 2?

Reply:
Removed.  This was old code.  The value is actually properly calculated in finalize (and there's a comment there).